### ISSUE TITLE
Gutenberg wraps Div->A->Div with P

### ISSUE THREAD

**Issue by ludenb** [WordPress/gutenberg#11038]
After Gutenberg 4.1 was released, it wraps `<div>` containig `<a>` containing `<div>` with `<p>` and duplicates `<a>` without `<div>` into one more `<p>`. There is no such a problem in Gutenberg 4.0

**To Reproduce**
1. Create a Custom HTML block with this HTML: `<div><a style="height:50px;width: 100px;background: #ddd;display: block;border-radius: 10px;"><div>Test link</div></a></div>`
2. Save and view the page
3. You will see 
![image](https://user-images.githubusercontent.com/12388514/47479652-1cbf4000-d871-11e8-930b-9ad4607a0e2f.png)

**Expected behavior**
![image](https://user-images.githubusercontent.com/12388514/47479617-03b68f00-d871-11e8-9305-08924675aa34.png)

**Desktop (please complete the following information):**
 - OS: Win7
 - Browser Chrome
 - Version 69.0.3497.100

**Comment by collinanderson** [comment]
I'm also having an issue with custom html getting wrapped in <p>'s that just started this morning with 4.1.0.

my post content has: (simplified)

```
<!-- wp:html -->
<a>
<div>
<h2></h2>
<p><span></span></p>
</div>
</a>
<!-- /wp:html -->
```

i'm getting this output:
```
<p><a></p>
<div>
<h2></h2>
<p><span></span></p>
</div>
<p></a></p>
```

**Comment by youknowriad** [comment]
It should be fixed by #11050
